File: cpe49題/10193 All You Need Is Love.cpp
#include <bits/stdc++.h>

using namespace std;

int toTen(string s1){
	int sum=0;
	int add=1;
	for(int i=s1.length()-1;i>=0;i--){
		if(s1[i]=='1'){
			sum+=add;
		}
		add=add*2;
	}
	return sum;
}

int loveFun(int x,int y){
	while(y!=0){
		int tmp=x%y;
		x=y;
		y=tmp;
	}
	return x;
}

int main(){
	int n,caseNum=1;
	cin>>n;
	while(n--){
		string s1,s2;		
		cin>>s1>>s2;
		
		int x=toTen(s1);
		int y=toTen(s2);
		
		int love=loveFun(x,y);
		
		if(love>1){
			cout<<"Pair #"<<caseNum<<": All you need is love!"<<endl;
		}else{
			cout<<"Pair #"<<caseNum<<": Love is not all you need!"<<endl;
		}
		caseNum++;
	}
}

File: cpe49題/10193 All You Need Is Love.py
import math

n=int(input())
caseNum=1

for i in range(n):
	n1=int(input(),2)
	n2=int(input(),2)
	
	ans=math.gcd(n1,n2)	
	isLove=False
	if ans>1:
		isLove=True
	
	if isLove:
		print(f"Pair #{caseNum}: All you need is love!")
	else:
		print(f"Pair #{caseNum}: Love is not all you need!")
		
	caseNum+=1
